### DISCUSSION TITLE
stockfish optimization(android)

### DISCUSSION THREAD

**Discussion by GustavoConde22** [official-stockfish/Stockfish#3783] [Q&A] [answered]
when the engine is on it suggests some possible moves, is it possible an option for the engine to calculate only a main line? an option to use all the memory to find only the best move! I have this doubt because my nodes/s are relatively low!

**Comment by vondele** [comment, marked as answer]
these are all features that can be controlled by the GUI / App you are using, the chess engine itself provides to the GUI means to set the number of threads used, and the amount of memory used.

**Reply by GustavoConde22** [reply]
got it! about the hash tables, my device has available 3GB of ram, what would be the proper size for the hash table?

**Reply by vondele** [reply]
2GB

**Reply by GustavoConde22** [reply]
thank you very much for the explanations!
